File: branches/hornetq-examples/src/main/java/deng/hornetqexamples/jms/SendToQueue.java
package deng.hornetqexamples.jms;

import javax.jms.Connection;
import javax.jms.ConnectionFactory;
import javax.jms.MessageProducer;
import javax.jms.Queue;
import javax.jms.Session;
import javax.jms.TextMessage;
import javax.naming.Context;
import javax.naming.InitialContext;

public class SendToQueue {
	
	public static void main(String[] args) throws Exception {
		SendToQueue bean = new SendToQueue();
		String queueName = args[0];
		String text = args[1];
		bean.send(queueName, text);
	}
	
	public void send(String queueName, String text) throws Exception {
		Connection connection = null;
		Context ctx = new InitialContext();
		try {
			ConnectionFactory cf = (ConnectionFactory)ctx.lookup("/ConnectionFactory");
			connection = cf.createConnection();
			Queue queue = (Queue)ctx.lookup(queueName);
			Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
			MessageProducer producer = session.createProducer(queue);
			TextMessage message = session.createTextMessage(text);
			System.out.println("Sending text: " + text);
			producer.send(message);		
			session.close();
			System.out.println("Message sent to queue: " + queueName);
		} finally {
			if (connection != null) {
				connection.close();
			}
			if (ctx != null) {
				ctx.close();
			}
		}
	}
}

File: branches/jboss-projects/scripts/sendToQueue.scala

//Zemian@Zemian-PC /jbs
//$ mkcpjb '/apps/hornetq-2.0.0.GA/lib/*'
//export CLASSPATH="C:\apps\jboss-5.1.0.GA\client\*;C:\apps\common\lib\*;target\dependency\*;target\classes;.\;C:\apps\hornetq-2.0.0.GA\lib\*"

/*
Zemian@Zemian-PC /jbs
$ scalac -cp $CLASSPATH -d "target\classes" sendToQueue.scala 

Zemian@Zemian-PC /jbs
$ scala -cp $CLASSPATH SendToQueue ExampleQueue "just a test"
$ scala -cp $CLASSPATH SendToQueue ExampleQueue --file "C:\cygwin\tmp\test-input.txt"

*/

object SendToQueue {

	import javax.naming._
	import javax.jms._
	
	// Listing a context names binding
	def send(qname : String, text : String) {
		val ctx = new InitialContext()
		val cf = ctx.lookup("ConnectionFactory").asInstanceOf[ConnectionFactory]
		val conn = cf.createConnection()
		val session = conn.createSession(false, Session.AUTO_ACKNOWLEDGE)
		val queue = session.createQueue(qname)
		val producer = session.createProducer(queue)
		val msg = session.createTextMessage(text)
		
		producer.send(msg)
		
		session.close
		conn.close
	}
	
	def readFile(fname : String) = scala.io.Source.fromString(fname).getLines().mkString

	def main(args : Array[String]) {
		val (opts, args2) = args.partition { e => e.startsWith("--") }
		val qname = args2(0)
		val text = if (opts.exists { e => e == "--file" }) readFile(args2(1)) else args2(1)
		send(qname, text)
		println("msg sent.")
		println(text)
	}
}
